### PR TITLE
Feature: Improved visual elements in the content area

### PR DESCRIPTION
**Pull request by 0x5bfa**
### Motivation & Context

- Improved DataGridHeader control
- Improved divider controls in High Contrast (red to white in dark HC mode)
- Improved divider margin

### PR Checklist

- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
  Closes https://github.com/files-community/Files/issues/13077
  Closes https://github.com/files-community/Files/issues/13078
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?

### Screenshots

_None_

### CODE REVIEW

**Comment by mdtauk** [comment]
image

**Comment by mdtauk** [comment]
High Contrast
image

**Comment by mdtauk** [comment]
> image !
>

The text should not have a backplate in High Contrast.

Some time ago the WinUI team took steps to fix these issues
https://github.com/microsoft/microsoft-ui-xaml/pull/1663#issuecomment-557858905

**Comment by 0x5bfa** [comment]
There's a lot of backplates in Files. that improvement should be out of the scope but I can if yair'd like.

**Comment by mdtauk** [comment]
> There's a lot of backplates in Files. that improvement should be out of the scope but I can if yair'd like.
> 
> image
> 
> Adding `HighContrastAdjustment="None"` will fix, then?

I am not sure of the technical detail, only that it was a change made in preparation for Windows 11.  

https://learn.microsoft.com/en-gb/uwp/api/windows.ui.xaml.uielement.highcontrastadjustment?view=winrt-22621

The change should be made across most of Files' UI in High Contrast

**Comment by yaira2** [comment]
@0x5bfa can you link the issue/feature request for this PR?

**Comment by 0x5bfa** [comment]
- [ ] Adjust placements of datagrid.
- [ ] See that was really fixed.

File: src/Files.App/Views/PaneHolderPage.xaml [hunk 1]
```diff
@@ -122,7 +122,7 @@
 				<icore:ChangePropertyAction
 					PropertyName="CurrentInstanceBorderBrush"
 					TargetObject="{Binding ElementName=PaneLeft}"
-					Value="{ThemeResource ControlStrokeColorDefault}" />
+					Value="{ThemeResource DividerStrokeColorDefaultBrush}" />
```
Comment (yaira2) [inline src/Files.App/Views/PaneHolderPage.xaml:125]: Is this correct? I think we need a custom brush so that we have the correct color in each theme.

File: src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml [hunk 2]
```diff
@@ -1079,11 +1080,11 @@
 										ToolTipService.ToolTip="{x:Bind AsGitItem.GitLastCommitFullSha, Mode=OneWay}"
 										Visibility="{Binding ColumnsViewModel.GitLastCommitShaColumn.Visibility, ElementName=PageRoot, Mode=OneWay}" />
 
-									<!--  Tags That Item Is Labeled  -->
+									<!--  Added Tags  -->
 									<Grid
 										x:Name="ItemTagGrid"
 										Width="{Binding ColumnsViewModel.TagColumn.LengthIncludingGridSplitter.Value, ElementName=PageRoot, Mode=OneWay}"
-										Padding="12,0,12,0"
+										Padding="10,0,0,0"
```
Comment (yaira2) [inline src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml:1087]: This is good, thanks!

File: src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml [hunk 3]
```diff
@@ -974,7 +975,7 @@
 									<!--  Item Name  -->
 									<Grid
 										Width="{Binding ColumnsViewModel.NameColumn.LengthIncludingGridSplitter.Value, ElementName=PageRoot, Mode=OneWay}"
-										Padding="12,0,0,0"
+										Padding="10,0,0,0"
```
Comment (yaira2) [inline src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml:978]: This should be 12, otherwise it doesn't line up with the header. That said, let's move the alignment changes into another PR so that we can focus on the divider brushes.

File: src/Files.App/App.xaml [hunk 4]
```diff
@@ -2,7 +2,8 @@
 <Application
 	x:Class="Files.App.App"
 	xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
-	xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
+	xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+	HighContrastAdjustment="None">
```
Comment (yaira2) [inline src/Files.App/App.xaml:6]: What does this property do?

**Comment by Jay-o-Way** [comment]
FYI

> **Note**
> If you do use measurement values, **all dimensions, margins, and padding should be in increments of 4 epx**. When XAML uses [effective pixels and scaling](https://learn.microsoft.com/windows/apps/design/layout/screen-sizes-and-breakpoints-for-responsive-design) to make your app legible on all devices and screen sizes, it scales UI elements by multiples of 4. Using values in increments of 4 results in the best rendering by aligning with whole pixels.

https://learn.microsoft.com/windows/apps/design/layout/alignment-margin-padding

**Comment by yaira2** [comment]
@Jay-o-Way we're well aware of this recommendation and we follow it around 95% of the time, but it's not always possible and it doesn't always make sense. As a side point, it's worth mentioning that according to WinUI, this rule doesn't apply anymore but we still follow it for consistency.

File: src/Files.App/App.xaml (hunk 4 above)
Comment (mdtauk) [reply src/Files.App/App.xaml:6]: > What does this property do?

https://learn.microsoft.com/en-gb/uwp/api/windows.ui.xaml.uielement.highcontrastadjustment?view=winrt-22621

File: src/Files.App/Views/PaneHolderPage.xaml (hunk 1 above)
Comment (0x5bfa) [reply src/Files.App/Views/PaneHolderPage.xaml:125]: I've confirmed that brush is appropriate in light(light dark), dark(light dark), high contrast(white). But previous one was red in hc theme. but we probably need to wrap this brush like App.Theme.DividerStrokeColorDefaultBrush.

File: src/Files.App/App.xaml (hunk 4 above)
Comment (0x5bfa) [reply src/Files.App/App.xaml:6]: written in the remarks section.

**Comment by 0x5bfa** [comment]
> FYI
> 
> Note If you do use measurement values, **all dimensions, margins, and padding should be in increments of 4 epx**. When XAML uses [effective pixels and scaling](https://learn.microsoft.com/windows/apps/design/layout/screen-sizes-and-breakpoints-for-responsive-design) to make your app legible on all devices and screen sizes, it scales UI elements by multiples of 4. Using values in increments of 4 results in the best rendering by aligning with whole pixels.
> 
> https://learn.microsoft.com/windows/apps/design/layout/alignment-margin-padding

I had to adjust position of the listview item texts to correspoing grid header texts position. why 10px is gridsplitter have 4 width. so i needed to reduce 2px from both sides.

**Comment by mdtauk** [comment]
> FYI
> 
> Note If you do use measurement values, **all dimensions, margins, and padding should be in increments of 4 epx**. When XAML uses [effective pixels and scaling](https://learn.microsoft.com/windows/apps/design/layout/screen-sizes-and-breakpoints-for-responsive-design) to make your app legible on all devices and screen sizes, it scales UI elements by multiples of 4. Using values in increments of 4 results in the best rendering by aligning with whole pixels.
> 
> https://learn.microsoft.com/windows/apps/design/layout/alignment-margin-padding

I have been sure to adhere to this with the designs.  Occasionally like with the sidebar items, we have 2px for top and bottom padding - so when stacked vertically, you get a 4px gap between items.  So you will see raw values in xaml that are not in multiples of 4, but when placed into the layouts, will in fact, meet this expectation.

**Comment by 0x5bfa** [comment]
image

File: src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml (hunk 3 above)
Comment (0x5bfa) [reply src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml:979]: Fixed

File: src/Files.App/App.xaml (hunk 4 above)
Comment (0x5bfa) [reply src/Files.App/App.xaml:6]: @mdtauk adding it in the app.xaml doesn't work. so added each page, i confirmed they are at work.

Comment (0x5bfa) [reply src/Files.App/App.xaml:6]: image

File: src/Files.App/Views/PaneHolderPage.xaml (hunk 1 above)
Comment (yaira2) [reply src/Files.App/Views/PaneHolderPage.xaml:125]: > but we probably need to wrap this brush like App.Theme.DividerStrokeColorDefaultBrush.

That was my assumption

File: src/Files.App/App.xaml (hunk 4 above)
Comment (yaira2) [reply src/Files.App/App.xaml:6]: Let's hold off on this change, I think we need to do some testing to see how it affects the rest of the app.

File: src/Files.App/App.xaml [hunk 5]
```diff
@@ -26,6 +27,10 @@
 			<CornerRadius x:Key="GridViewThumbnailCornerRadius">2</CornerRadius>
 			<CornerRadius x:Key="DetailsLayoutThumbnailCornerRadius">2</CornerRadius>
 
+			<Style TargetType="FlyoutPresenter">
+				<Setter Target="HighContrastAdjustment" Value="None" />
+			</Style>
```
Comment (yaira2) [inline src/Files.App/App.xaml:32]: Let's hold off on this change.

File: src/Files.App/App.xaml (hunk 4 above)
Comment (0x5bfa) [reply src/Files.App/App.xaml:6]: we can test in this PR. This PR's purpose is to improve visual elements and so I think it is not out of the scope of the PR.

Comment (0x5bfa) [reply src/Files.App/App.xaml:6]: And I tested in the contents area and the settings area. The controls from MUXC like TabView has already applied this property in the custom style ([custom TabView style](https://github.com/files-community/Files/blob/main/src/Files.App/UserControls/MultitaskingControl/HorizontalMultitaskingControl.xaml#L218)).

File: src/Files.App/App.xaml (hunk 5 above)
Comment (0x5bfa) [reply src/Files.App/App.xaml:32]: Actually I fixed #13128 here lol

Comment (0x5bfa) [reply src/Files.App/App.xaml:32]: Sure!

File: src/Files.App/App.xaml (hunk 4 above)
Comment (yaira2) [reply src/Files.App/App.xaml:6]: > @mdtauk adding it in the app.xaml doesn't work. so added each page, i confirmed they are at work.

This seems to be a bug, do you know if it's reported on the WinUI repo yet?
